nototools/android_patches.py

```diff
@@ -0,0 +1,431 @@
+  Emoji font when a serif style was specified for text, leading to
+  inconsistent use of emoji in sans versus serif text.
+
+  Bug: 24740612"""
+
+  CHARS_TO_DELETE = {
+      0x263A, # WHITE SMILING FACE
+      0x2660, # BLACK SPADE SUIT
+      0x2663, # BLACK CLUB SUIT
+      0x2665, # BLACK HEART SUIT
+      0x2666, # BLACK DIAMOND SUIT
+  }
+  for font_file in glob.glob(path.join(SRC_DIR, 'NotoSerif-*.ttf')):
+    print 'delete smiley/playing card', font_file
+    out_file = path.join(DST_DIR, path.basename(font_file))
+    subset.subset_font(font_file, out_file, exclude=CHARS_TO_DELETE)
+
+
+def _patch_hyphen_armenian_ethiopic():
+  """Add hyphen-minus glyphs to Armenian and Ethiopic fonts.
+
+  This is to enable Armenian and Amharic langauges to be hyphenated
+  properly, since Minikin's itemizer currently shows tofus if a an
+  automatically hyphenated word is displated in a font that has neither
+  HYPHEN nor HYPHEN-MINUS.
+
+  In practice only U+002D HYPHEN-MINUS is added, since Noto LGC fonts
+  don't have U+2010 HYPHEN.)
+
+  Bug: 21570828"""
+
+  FONTS = [
+      'NotoSansArmenian-Regular.ttf',
+      'NotoSansArmenian-Bold.ttf',
+      'NotoSerifArmenian-Regular.ttf',
+      'NotoSerifArmenian-Bold.ttf',
+      'NotoSansEthiopic-Regular.ttf',
+      'NotoSansEthiopic-Bold.ttf',
+  ]
+
+  HYPHENS = {0x002D, 0x2010}
+
+  for font_name in FONTS:
+    lgc_font_name = (
+        font_name.replace('Armenian', '')
+        .replace('Ethiopic', ''))
+
+    font_file = path.join(SRC_DIR, font_name)
+    lgc_font_file = path.join(SRC_DIR, lgc_font_name)
+
+    chars_to_add = (
+        (HYPHENS - coverage.character_set(font_file))
+        & coverage.character_set(lgc_font_file))
+
+    if chars_to_add:
+      print 'patch hyphens', font_name
+      merger.merge_chars_from_bank(
+          path.join(SRC_DIR, font_name),
+          path.join(SRC_DIR, lgc_font_name),
+          path.join(DST_DIR, font_name),
+          chars_to_add)
+
+
+# Characters we have decided we are doing as emoji-style in Android,
+# despite UTR#51's recommendation
+ANDROID_EMOJI = [
+    0x2600, # ☀ BLACK SUN WITH RAYS
+    0x2601, # ☁ CLOUD
+    0X260E, # ☎ BLACK TELEPHONE
+    0x261D, # ☝ WHITE UP POINTING INDEX
+    0x263A, # ☺ WHITE SMILING FACE
+    0x2660, # ♠ BLACK SPADE SUIT
+    0x2663, # ♣ BLACK CLUB SUIT
+    0x2665, # ♥ BLACK HEART SUIT
+    0x2666, # ♦ BLACK DIAMOND SUIT
+    0x270C, # ✌ VICTORY HAND
+    0x2744, # ❄ SNOWFLAKE
+    0x2764, # ❤ HEAVY BLACK HEART
+]
+
+
+def _remove_cjk_emoji():
+  """(two fixes)
+  ---
+  Remove default emoji characters from CJK fonts.
+
+  Twenty-six characters that Unicode Technical Report #51 "Unicode
+  Emoji" defines as defaulting to emoji styles used to be displayed as
+  black and white ("text" style) before this. This patch removes those
+  characters from Noto CJK fonts, so they get displayed as color.
+
+  As a side effect of the subsetting process, some ideographic
```

Comment:
Please make sure the variation sequences are not lost anymore, now that we support them.